### PR TITLE
Bug fix and bring up to date `editable` method

### PR DESCRIPTION
**Pull request by timbl**
Remove a kludge which was put in to counter a long gone NSS bug. 

Also fix a bug with getting editable status of a file which is 404 - ie  would the file be editable if it existed?

### CODE REVIEW

File: src/update-manager.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@
 ** 2010-08-08 TimBL folded in Kenny's WEBDAV
 ** 2010-12-07 TimBL addred local file write code
 */
+/* source-trace 20209521a */
```
Comment (megoth) [inline src/update-manager.ts:7]: Do we need this in master?

File: src/update-manager.ts [hunk 2]
```diff
@@ -769,7 +758,11 @@ export default class UpdateManager {
         (this.store.fetcher.load(doc) as Promise<Response>).then(response => {
           this.update(deletions, insertions, callback, true)
         }, err => {
-          throw new Error(`Update: Can't read ${doc} before patching: ${err}`)
+            if (err.response.status === 404) { // nonexistent files are fine
+              this.update(deletions, insertions, callback, true)
```
Comment (megoth) [inline src/update-manager.ts:762]: Maybe create a test to cover this branch?

Comment (timbl) [reply src/update-manager.ts:761]: Done

File: src/update-manager.ts (hunk 1 above)
Comment (timbl) [reply src/update-manager.ts]: Removed

**Pull request merged**